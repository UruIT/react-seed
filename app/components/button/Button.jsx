import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

const log = e => console.warn(e.target.textContent);

const Button = ({ text, onClick }) => (
	<div className={styles.container} onClick={onClick} >
		{ text }
	</div>
);

Button.propTypes = {
	onClick: PropTypes.func,
	text: PropTypes.string
};
Button.defaultProps = {
	onClick: log,
	text: '...'
}

export default Button;
