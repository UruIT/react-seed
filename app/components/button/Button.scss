.container {
	border: 1px solid black;
}
