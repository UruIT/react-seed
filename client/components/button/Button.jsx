import React from 'react';
import PropTypes from 'prop-types';
import styles from './button.scss';

const show = e => alert(e.target.textContent);

const Button = ({ text, onClick }) => (
	<div className={styles.container} onClick={onClick}>
		{ text }
	</div>
);

Button.propTypes = {
	onClick: PropTypes.func,
	text: PropTypes.string
};
Button.defaultProps = {
	onClick: show,
	text: '...'
};

export default Button;
