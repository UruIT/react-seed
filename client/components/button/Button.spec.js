import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<Button/>', () => {
	it('snapshot - default props', () => {
		const tree = renderer.create(<Button />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('snapshot - with TEXT', () => {
		const tree = renderer.create(<Button text="TEXT" />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('Button - on Click', () => {
	it('should call onClick callback', () => {
		const onClickCb = jest.fn();

		const app = shallow(<Button onClick={onClickCb} />);

		app.simulate('click');

		expect(onClickCb).toBeCalled();
	});
});
