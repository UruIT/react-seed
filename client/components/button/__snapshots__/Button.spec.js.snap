// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> with default props 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  ...
</div>
`;

exports[`<Button/> with some props 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  TEXT
</div>
`;
